File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        vals = {}
        
        for i,num in enumerate(nums):
            if target - num in vals:
                return [i, vals[target -num]]
            else:
                vals[num] = i
                
        return [-1, -1]
        

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    // O(n) time and space
    vector<int> twoSum(vector<int>& nums, int target) {
        
        // hashmap in C++
        std::unordered_map<int, int> map; 
        
        for (int i = 0; i < nums.size(); i++) {
            if (map.find(nums[i]) != map.end()) {
                cout << "hello there" << endl;
                // prints iterator, gives compile error, need to get the value itself
                // cout << map.find(nums[i]) << endl;
                return {i, map[nums[i]]};
            } else {
                map[target - nums[i]] = i;
            }
        }
        
        // printing the map, commented out to improve runtime
        // for (const auto& pair : map) {
        //     std::cout << "Key: " << pair.first << ", Value: " << pair.second << std::endl;
        // }

        // default
        return {-1, -1};
    }
};